duckdb_test.go

```diff
@@ -41,15 +42,38 @@ func TestOpen(t *testing.T) {
 	})
 
 	t.Run("with invalid config", func(t *testing.T) {
-		db, _ := sql.Open("duckdb", "?threads=NaN")
-		err := db.Ping()
+		_, err := sql.Open("duckdb", "?threads=NaN")
```

Comment:
the behaviour of `sql.Open` has changed a bit here. Earlier it would just accept `dsn` as it is without validating anything and would throw error on `Ping()`. But now it will throw err on call to this method.  Docs say 
> // Open may just validate its arguments without creating a connection
// to the database. To verify that the data source name is valid, call
// Ping.

So it seems acceptable otherwise we may also create `C.duckdb_database` on call to `Connector.Connect` store it and then subsequent calls use it without creating a new one.

Reply:
I agree, it seems acceptable to fail already on `Open` – and it makes the code simpler